### ISSUE TITLE
Demo app

### ISSUE THREAD

**Issue by villander** [bgentry/ember-apollo-client#62]
We need of demo/dummy app as [ember-power-select](http://www.ember-power-select.com/), so we will have a case, besides making better the experience of each user that uses this addon with docs, cookbook and others

**Comment by bgentry** [comment]
Yes, this would be really great. I haven't found the time to spend on it so far but it's also an area where anybody would be able to contribute, especially in terms of getting the app skeleton built & designed.

**Comment by dfreeman** [comment]
It's not _quite_ ready for primetime yet, but https://ember-learn.github.io/ember-cli-addon-docs may be worth being aware of for this

**Comment by villander** [comment]
A demo/dummy app with authentication using [Github API](https://developer.github.com/v4/). Would be a good case, especially if we have unit tests of integration and acceptance.

**Comment by bgentry** [comment]
Agreed!

Good test examples including pretender and schema mocking would be helpful. I also recently set up my full schema with VS Code and a `.gqlconfig`, that was a pretty magical improvement.

**Comment by villander** [comment]
Acceptance tests with pretender and schema or mirage is somewhat verbose, you need to track a lot. It is best to use acceptance tests like E2E. We do not have an ergonomic tool to mock GraphQL on the front, like the mirage.

**Comment by bgentry** [comment]
I opened #106 to track creating our own documentation site, which could potentially be the same as the demo app.

@villander re: mocking, you are familiar with the built-in Apollo Client mocking functions, right?

https://dev-blog.apollodata.com/mocking-your-server-with-just-one-line-of-code-692feda6e9cd
https://www.apollographql.com/docs/graphql-tools/mocking.html
https://medium.com/@carlos_42328/mocking-api-responses-with-react-apollo-11eb4610debe

It's indeed a bit verbose for tests where you need to really customize the response. Would love to see a better tool for that.
